src/components/List/List.tsx

```diff
@@ -62,22 +69,25 @@ class List extends UIComponent<Extendable<ListProps>, ListState> {
     selection: PropTypes.bool,
     truncateContent: PropTypes.bool,
     truncateHeader: PropTypes.bool,
+    selectedItemIndex: PropTypes.number,
+    defaultSelectedItemIndex: PropTypes.number,
   }
 
   static defaultProps = {
     as: 'ul',
     accessibility: listBehavior as Accessibility,
   }
 
+  static autoControlledProps = ['selectedItemIndex']
+  getInitialAutoControlledState() {
+    return { selectedItemIndex: -1, focusedItemIndex: 0 }
```

Comment:
currently it initializes more than just auto controlled state (`focusedItemIndex`). we can use the following to initialize these guys:

```jsx
 static defaultProps = {
    as: 'ul',
    selectedItemIndex: -1,
    accessibility: listBehavior as Accessibility,	    accessibility: listBehavior as Accessibility,
  }	  
```
and initialize `state.focusedItemIndex` in constructor